src/transformers/models/seamless_m4t/configuration_seamless_m4t.py

```diff
@@ -0,0 +1,415 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" SeamlessM4T model configuration"""
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+SEAMLESS_M4T_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "ylacombe/hf-seamless-m4t-medium": "https://huggingface.co/ylacombe/hf-seamless-m4t-medium/resolve/main/config.json",
+    # See all SeamlessM4T models at https://huggingface.co/models?filter=seamless_m4t
+}
+
+
+class SeamlessM4TConfig(PretrainedConfig):
```

Comment:
I believe that the config docstring is looking good here: https://moon-ci-docs.huggingface.co/docs/transformers/pr_25693/en/model_doc/seamless_m4t#transformers.SeamlessM4TConfig

Also some other reasons behind my choice:
- the code is already pretty heavy and adding nested configs would complicate even more
- Some parameters are shared and it would make a lot of redundancies
- Sub-models are not supposed to be used alone, so IMO it won't add that much added values to have nested configs here

Of course, I'm open to suggestions and I can change my decision if you want.

In the meantime, I'll try to hardcode some parameters, based on relevancy.